src/game/behaviors/clock_arm.inc.c

```diff
@@ -3,20 +3,93 @@
  */
 
 #include "sm64ap.h"
+#include "../../sm64ap.h"
 
 void bhv_rotating_clock_arm_loop(void) {
-    if (o->oBehParams != SM64AP_CourseToTTC()) {
+    int entrance = SM64AP_EntranceToTTC();
+    int course = entrance / 10;
+    s32 isMinuteHand = cur_obj_has_behavior(bhvClockMinuteHand);
+    u16 rollAngle = o->oFaceAngleRoll;
+    if ((o->oBehParams <= LEVEL_UNKNOWN_38 && o->oBehParams != course) ||
+        (o->oBehParams > LEVEL_UNKNOWN_38 && o->oBehParams != entrance)) { // Special handling is done for THI, which has two entrances. If param is greater than max level id, check directly against entrance
+        // This entrance doesn't lead to TTC. Remove.
+        obj_mark_for_deletion(o);
+        return;
+    } else if (o->oBehParams == LEVEL_VCUTM && !SM64AP_MoatDrained()) { // If the moat isn't drained the clock hands stick out of the water. Remove them.
         obj_mark_for_deletion(o);
         return;
-    } else {
+    } else if (isMinuteHand && o->oAction < 2) {
+        // This is the correct clock hand. Set the clock action
         SM64AP_SetClockToTTCAction(&(o->oAction));
     }
-    
     struct Surface *marioSurface;
-    u16 rollAngle = o->oFaceAngleRoll;
     o->oFloorHeight =
         find_floor(gMarioObject->oPosX, gMarioObject->oPosY, gMarioObject->oPosZ, &marioSurface);
-
+    // On first frame, set entrance-specific settings
+    if (o->oTimer == 0) {
+        switch (o->oBehParams) {
```

Comment:
Just wanted to say, scaling the clock hands is great :)
And, differently from the reflections, it's an entirely local change